src/System.Private.CoreLib/src/Internal/Runtime/InteropServices/IsolatedComponentLoadContext.cs

```diff
@@ -0,0 +1,43 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Reflection;
+using System.Runtime.Loader;
+
+namespace Internal.Runtime.InteropServices
```

Comment:
This was previously a nicely hidden away type in the COM scenario, which is why there wasn't any comments/documentation for it. Now that is is being reused it would be nice for people to know why it was created and when it should be used.